services/blockchain-service/blockchain-service-base.js

```diff
@@ -220,6 +245,63 @@ class BlockchainServiceBase {
         };
     }
 
+    async waitForTransactionFinalization(initialReceipt, blockchain) {
+        await this.ensureBlockchainInfo(blockchain);
+        const web3Instance = await this.getWeb3Instance(blockchain);
+
+        const startTime = Date.now();
+        let reminingTime = 0;
+        let receipt = initialReceipt;
+        let finalized = false;
+      
+        try {
+            while (!finalized && (Date.now() - startTime + reminingTime) < blockchain.transactionFinalityMaxWaitTime) {
+                try {
+                    // Check if the block containing the transaction is finalized
+                    const finalizedBlockHash = await web3Instance.chain.getFinalizedHead();
+                    const finalizedBlock = (await web3Instance.chain.getBlock(finalizedBlockHash)).block;
+                    const finalizedBlockNumber = Web3.utils.hexToNumber(finalizedBlock.header.number);
+                    if (finalizedBlockNumber >= receipt.blockNumber) {
```

Comment:
Would it be usefull to add here check if tx is in finalized block

Reply:
But that's exactly what it does?